File: src/main/java/com/protryon/jasm/Field.java
package com.protryon.jasm;

public class Field {

    public final Klass parent;
    public JType type;
    public String name;
    public boolean isPublic = false;
    public boolean isPrivate = false;
    public boolean isProtected = false;
    public boolean isStatic = false;
    public boolean isFinal = false;
    public boolean isVolatile = false;
    public boolean isTransient = false;
    public boolean isSynthetic = false;
    public boolean isEnum = false;
    public Constant constantValue;

    // set for created methods in things like stdlib or unincluded libs
    public boolean isDummy = false;

    public Field(Klass parent, JType type, String name) {
        this.parent = parent;
        this.type = type;
        this.name = name;
    }

}

File: src/main/java/com/protryon/jasm/Field.kt
package com.protryon.jasm

class Field(val parent: Klass, var type: JType, var name: String) {
    var isPublic = false
    var isPrivate = false
    var isProtected = false
    var isStatic = false
    var isFinal = false
    var isVolatile = false
    var isTransient = false
    var isSynthetic = false
    var isEnum = false
    var constantValue: Constant<*>? = null

    // set for created methods in things like stdlib or unincluded libs
    var isDummy = false

}
